### PR TITLE
Code cleanup and optimisations

### PR DESCRIPTION
**Pull request by kaivan-s**
Following changes are incorporated in the PR
1. Code cleanup
2. Code optimisations

### CODE REVIEW

File: Models/Arima.py
Comment (VanshChiri) [inline Models/Arima.py:1]: data.sort_index() looks repetitive

Comment (kaivan-s) [reply Models/Arima.py:1]: Not sorting for all files currently and hence kept separately. Once we finalise the models, we will do move it in preprocessing step only.

File: DL_Models/LSTM.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    def forward(self, x):
+        h0 = torch.zeros(self.num_layers, x.size(0), self.hidden_size).requires_grad_()
+        c0 = torch.zeros(self.num_layers, x.size(0), self.hidden_size).requires_grad_()
+
+        out, (hn, cn) = self.lstm(x, (h0.detach(), c0.detach()))
+        out = self.fc(out[:, -1, :])
+        return out
+
+
+def min_max_scaler(data):
```
Comment (git-shrikant-sharma) [inline DL_Models/LSTM.py:26]: Have we used the normalised scaling (v(alue-mean)/std. dev.) anywhere?

File: DL_Models/LSTM.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        a = dataset[i:(i + look_back), :]
+        dataX.append(a)
+        dataY.append(dataset[i + look_back, 0])
+    return np.array(dataX), np.array(dataY)
+
+
+def execute_lstm(data):
+    data = data.reset_index()
+    data.drop(columns=['Date', 'Year', 'Month', 'Day', 'Season'], inplace=True)
+    scaled_data, data_min, data_max = min_max_scaler(data.values)
```
Comment (kaivan-s) [inline DL_Models/LSTM.py:48]: Calling the function here !

File: Execute/execute.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from Models.ETS import execute_ets
+from Models.Arima import execute_arima
+
+from Feature_Selection.Time_Series_Decomposition import evaluate_time_series_decomposition
+from Feature_Selection.Correlation_Analysis import evaluate_correlation_analysis
+from Feature_Selection.Tree_Based_Models import evaluate_tree_based_models
+from Feature_Selection.Recursive_Feature_Elimination import evaluate_recursive_feature_elimination
+from DL_Models.LSTM import execute_lstm
+
+data = pd.read_csv("../Data/ICAC multiple variables.csv", parse_dates=['Date'], dayfirst=True)
```
Comment (git-shrikant-sharma) [inline Execute/execute.py:13]: Are we planning to have a filename as an argument and then the system reads that file?

File: DL_Models/LSTM.py (hunk 1 above)
Comment (kaivan-s) [reply DL_Models/LSTM.py:26]: In line 48

File: Execute/execute.py (hunk 3 above)
Comment (kaivan-s) [reply Execute/execute.py:13]: Yes something like that once we do the s3 changes and also some UI is developed.

File: DL_Models/LSTM.py (hunk 1 above)
Comment (git-shrikant-sharma) [reply DL_Models/LSTM.py:26]: line 48 doesn't have standard deviation I think. Am I missing something?

Comment (kaivan-s) [reply DL_Models/LSTM.py:26]: I think what you mean is standardization and not normalization. Both are different scaling methods.

Comment (git-shrikant-sharma) [reply DL_Models/LSTM.py:26]: Yes, right. Standardization is also something that would be relevant to us, I feel. What do you think?

Comment (kaivan-s) [reply DL_Models/LSTM.py:26]: Creating an issue for that. Not taking it up as a part of this PR since it's for code cleanup !

**Review by git-shrikant-sharma — APPROVED** [review]

File: DL_Models/LSTM.py (hunk 1 above)
Comment (kaivan-s) [reply DL_Models/LSTM.py:26]: https://github.com/Nexus-Optima/Backend/issues/8 Added an issue.
